### PR TITLE
cli: implement correct self-updating

### LINKED ISSUE #7107: "Permission denied (os error 13)" during update
I am running code-server on Ubuntu 20.04.4 LTS and got an error during the update procedure:

```
$ code-server --verbose update
[2022-08-23 12:37:29] debug Server args: CodeServerArgs { host: None, port: None, socket_path: None, telemetry_level: None, log: None, accept_server_license_terms: true, verbose: true, install_extensions: [], uninstall_extensions: [], list_extensions: false, show_versions: false, category: None, pre_release: false, force: false, start_server: false, connection_token: None, connection_token_file: None, without_connection_token: false }
[2022-08-23 12:37:30] info Update to 20220823.4 (UTC 1661221684) available
[2022-08-23 12:37:30] error failed to create file: Permission denied (os error 13)
```

code-server runed from home dir under unprivileged user. This user owns all code-server's dirs. 

all works fine via sudo:
```
$ sudo code-server --verbose update
[2022-08-23 12:41:39] debug Server args: CodeServerArgs { host: None, port: None, socket_path: None, telemetry_level: None, log: None, accept_server_license_terms: true, verbose: true, install_extensions: [], uninstall_extensions: [], list_extensions: false, show_versions: false, category: None, pre_release: false, force: false, start_server: false, connection_token: None, connection_token_file: None, without_connection_token: false }
[2022-08-23 12:41:40] info Update to 20220823.4 (UTC 1661221684) available
[2022-08-23 12:41:48] info Update complete
```
What directories are used during the update? 

What permissions do I need to set?

Thank you!

### LINKED ISSUE #159798: Update CLI's self-updating functionality
The CLI's self-updating functionality to be disabled when installed as an integrated CLI, or pull from the update server when installed as a standalone CLI.

### LINKED ISSUE #161914: code-tunnel: license prompt text missing
Testing #161856 on Linux

image

### PR DESCRIPTION
**Pull request by connor4312**
This PR got kind of disparate as I went through fixing issues to make self-updating work...

- Start separating a "standalone" CLI. This is a little awkward with clap-derive, but I got it working. The CLI looks at the files around it to determine which interface to use
- Remove the old ad-hoc update code for code-server, and use the update service instead.
- Fix some of the "permission denied" errors people got while updating before. We need to rename the old running binary, not just overwrite it.
- Only expose the update method when running in "standalone" mode
- Build Windows 32 bit and fix issues with build variables not getting set correctly

Fixes https://github.com/microsoft/vscode/issues/159798
Fixes https://github.com/microsoft/vscode/issues/161914
Fixes https://github.com/microsoft/vscode-internalbacklog/issues/3192
Fixes https://github.com/microsoft/vscode-remote-release/issues/7107

### CODE REVIEW

**Comment by connor4312** [comment]
This is ready for review, but is stacked on the "build" PR and should _not_ be merged yet. Once the build PR is in, I will change the target of this one to `main`

File: cglicenses.json [hunk 1]
```diff
@@ -259,38 +259,6 @@
 			"THE SOFTWARE."
 		]
 	},
-	{
```
Comment (connor4312) [inline cglicenses.json:262]: We no longer have this dependency

File: build/azure-pipelines/cli/cli-compile-and-publish.yml [hunk 2]
```diff
@@ -8,26 +8,20 @@ parameters:
...
+  - script: cargo build --release --target ${{ parameters.VSCODE_CLI_TARGET }} --bin=code
     displayName: Compile ${{ parameters.VSCODE_CLI_TARGET }}
     workingDirectory: $(Build.SourcesDirectory)/cli
     env:
-      VSCODE_CLI_VERSION: $(VSCODE_CLI_VERSION)
-      VSCODE_CLI_REMOTE_LICENSE_TEXT: $(VSCODE_CLI_REMOTE_LICENSE_TEXT)
-      VSCODE_CLI_REMOTE_LICENSE_PROMPT: $(VSCODE_CLI_REMOTE_LICENSE_PROMPT)
-      VSCODE_CLI_ASSET_NAME: ${{ parameters.VSCODE_CLI_ARTIFACT }}
-      VSCODE_CLI_AI_KEY: $(VSCODE_CLI_AI_KEY)
-      VSCODE_CLI_AI_ENDPOINT: $(VSCODE_CLI_AI_ENDPOINT)
```
Comment (connor4312) [inline build/azure-pipelines/cli/cli-compile-and-publish.yml:20]: These were already set as environment variables, no need to re-add them here

File: build/azure-pipelines/cli/cli-compile-and-publish.yml [hunk 3]
```diff
@@ -8,26 +8,20 @@ parameters:
...
       ${{ each pair in parameters.VSCODE_CLI_ENV }}:
         ${{ pair.key }}: ${{ pair.value }}
 
   - ${{ if or(contains(parameters.VSCODE_CLI_TARGET, '-windows-'), contains(parameters.VSCODE_CLI_TARGET, '-darwin')) }}:
     - task: ArchiveFiles@2
       inputs:
         ${{ if contains(parameters.VSCODE_CLI_TARGET, '-windows-') }}:
-          rootFolderOrFile: $(Build.SourcesDirectory)/cli/target/${{ parameters.VSCODE_CLI_TARGET }}/release/code-tunnel.exe
+          rootFolderOrFile: $(Build.SourcesDirectory)/cli/target/${{ parameters.VSCODE_CLI_TARGET }}/release/code.exe
```
Comment (connor4312) [inline build/azure-pipelines/cli/cli-compile-and-publish.yml:22]: These changes (and most of the ones in the build script below) are related to renaming the standalone CLI from `code-tunnel` to `code`. It still gets bundled as `code-tunnel` in the desktop builds, but users who download the VS Code CLI should just get `code`.

File: cli/src/tunnels/code_server.rs [hunk 4]
```diff
@@ -238,7 +239,7 @@ pub enum AnyCodeServer {
 
 pub enum CodeServerOrigin {
 	/// A new code server, that opens the barrier when it exits.
-	New(Child),
+	New(Box<Child>),
```
Comment (connor4312) [inline cli/src/tunnels/code_server.rs:242]: This was just a recommended clippy change (on nightly), no functional changes

**Review by aeschli — APPROVED** [review]

**Pull request merged**